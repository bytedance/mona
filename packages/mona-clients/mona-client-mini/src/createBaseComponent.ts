import React from 'react';

export default function createBaseComponent<P>(name: string) {
  const Component = React.forwardRef<any, any>(({ children, ...props }, ref) =>
    React.createElement(name, { ...props, ref }, children),
  );
  Component.displayName = name;
  return Component as React.ComponentType<P>;
}
