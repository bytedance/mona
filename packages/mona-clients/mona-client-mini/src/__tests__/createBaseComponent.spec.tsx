import React from 'react';
import TestRenderer from 'react-test-renderer';

import createBaseComponent from '../createBaseComponent';

test('createBaseComponent', () => {
  const Test = createBaseComponent('test-test');
  const testIns = TestRenderer.create(<Test />);
  expect(testIns.toJSON()).toMatchSnapshot();
});
