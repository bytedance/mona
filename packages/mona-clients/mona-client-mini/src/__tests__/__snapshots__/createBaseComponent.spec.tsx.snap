// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createBaseComponent 1`] = `<test-test />`;

exports[`react ref 1`] = `<div />`;
